wagtail/contrib/search_promotions/views.py

```diff
@@ -19,16 +20,30 @@
 )
 @vary_on_headers('X-Requested-With')
 def index(request):
-    is_searching = False
-    query_string = request.GET.get('q', "")
+    # Ordering
+    valid_ordering = ['query_string', '-query_string', 'ohits', '-ohits']
+    ordering = valid_ordering[0]
+
+    if 'ordering' in request.GET and request.GET['ordering'] in valid_ordering:
+        ordering = request.GET['ordering']
+
+    # Query
+    queries = Query.objects.filter(editors_picks__isnull=False).distinct()
 
-    queries = Query.objects.filter(editors_picks__isnull=False).distinct().order_by('query_string')
+    if 'ohits' in ordering:
+        queries = queries.annotate(ohits=functions.Coalesce(Sum('daily_hits__hits'), 0))
+
+    queries = queries.order_by(ordering)
 
     # Search
+    is_searching = False
```

Comment:
The search behavior doesn't work quite as expected for me, because the AJAX call loses the ordering once you start typing.

The search behavior is a little inconsistent, although this appears to be common in other parts of Wagtail (for example the user search). I didn't see this behavior mentioned on #5191.

If you search first and then order, you lose the search term:

1. Search first, URL is http://localhost:8000/admin/searchpicks/?q=b.
2. Click on ordering, link goes to http://localhost:8000/admin/searchpicks/?ordering=ohits, losing the search.

If you order first and then search, you lose the order:

1. Order first, URL is http://localhost:8000/admin/searchpicks/?ordering=ohits.
2. Start typing a search, AJAX request is made to http://localhost:8000/admin/searchpicks/?q=b (without the ordering param), so you lose the order.

To me the expected behavior is that searches should definitely persist across ordering. I'm not sure about the other way, whether ordering should persist if you do a new search, but I think maybe it should, as the user has indicated a preferred ordering (in this case, a common one would probably be to order by most common search term).

This doesn't need to hold up this PR, but I'm curious what people think is the best way to handle this.

Reply:
@chosak  - thanks for that, I was aware of this issue and intentionally have not fixed it in this PR. I assumed, wrongly, an issue had already been created for this (I thought I had created one a year or so ago). I searched but could not find it.

I have now created a new issue that contains your notes and a bit more.

https://github.com/wagtail/wagtail/issues/5529